lib/matplotlib/backend_tools.py

```diff
@@ -77,14 +77,38 @@ def __init__(self, toolmanager, name):
                       'experimental for now, the API will likely change in ' +
                       'version 2.1, and some tools might change name')
         self._name = name
+        self._toolmanager = toolmanager
         self._figure = None
-        self.toolmanager = toolmanager
-        self.figure = toolmanager.canvas.figure
 
     @property
     def figure(self):
         return self._figure
 
+    @figure.setter
+    def figure(self, figure):
+        self.set_figure(figure)
+
+    @property
+    def canvas(self):
+        return self.figure.canvas
+
+    @property
+    def toolmanager(self):
+        return self._toolmanager
+
+    def set_figure(self, figure):
+        """
+        Called when the figure is setted
```

Comment:
Maybe "Assign a figure to the tool"?  Or omit; the function is obvious from the name.